workers/proto/kitchen_sink/kitchen_sink.proto

```diff
@@ -211,6 +211,30 @@ message ExecuteActivityAction {
   message ClientActivity {
       ClientSequence client_sequence = 1;
   }
+
+  // Activity that throws retryable errors for N attempts, then succeeds.
+  // Tests activity retry behavior.
+  message RetryableErrorActivity {
+    // How many attempts should fail before succeeding (1-indexed)
+    int32 fail_attempts = 1;
+  }
+
+  // Activity that runs too long for N attempts (causing timeout), then completes.
+  // Tests StartToCloseTimeout behavior with retries.
+  // On failing attempts, the activity runs until context cancellation (timeout).
+  // On success, it runs for duration < StartToCloseTimeout (runs for 0.5x StartToClose timeout).
+  message TimeoutActivity {
+    // How many attempts should timeout before succeeding (1-indexed)
+    int32 fail_attempts = 1;
+  }
+
+  // Activity that skips heartbeats for N attempts (causing heartbeat timeout), then sends them.
+  // Tests HeartbeatTimeout behavior with retries.
+  // Duration is derived from activity info's HeartbeatTimeout (runs for 2x heartbeat timeout).
```

Comment:
```suggestion
  // On success, runs for 2x heartbeat timeout.
```
To be consistent with the other one - but, again I would also probably just parameterize here.